tfdml/kernels/dml_diag_part_op.cc

```diff
@@ -104,17 +99,12 @@ class DmlDiagPartKernel : public DmlKernel
             0,
             0,
             0,
-            (out_num_elements + 1) * stride_multiplier,
+            (out_num_elements + 1),
         };
 
         DmlTensorInfo input;
         input.kernel_index = 0;
-        input.desc = DmlTensorDesc(
-            dtype_dml,
-            input_sizes,
-            input_strides,
-            0,
-            end_padding_in_bytes);
+        input.desc = DmlTensorDesc(dtype_dml, input_sizes, input_strides, 0, 0);
```

Comment:
Nit: you can remove the values for `guaranteed_base_offset_alignment` and `end_padding_in_bytes` since their default value is already 0.